### PR TITLE
Use mock http servers from okhttp

### PR DESCRIPTION
**Pull request by Semernitskaya**
## Description

- All tests in the project that used `com.github.restdriver.clientdriver.ClientDriver` as mock web server were migrated to `okhttp3.mockwebserver.MockWebServer` instead.
- `MockWebServerUtil` util class was introduced to simplify migration and add functionality missing in the new `okhttp3.mockwebserver.MockWebServer`.
- `MockWebServerUtil` class is duplicated across modules and can be reviewed only in one place. The same duplication approach was used for the existing `MockSetup` class. In the future we can avoid copying and pasting them by introducing a new project module that will be used only with the test scope (see discussion https://github.com/zalando/riptide/pull/1441#discussion_r1162980272).
- `com.github.restdriver.clientdriver.ClientDriver` allows you to define response for a **specific** request and then use `verify()` method to ensure that driver got this exact request. For `okhttp3.mockwebserver.MockWebServer` you usually define common response for all possible requests, then check the request you actually got (using `server.takeRequest(...)` method). Multiple `verify()` methods were added to `MockWebServerUtil` class to simplify such check. See https://github.com/zalando/riptide/pull/1441/files#diff-c2b11cfb53a03add49f5f20b1efdbd3c238230a2ec547eed41218d6c8dbdfff6R95 for a simple example of old validation vs new validation. 

## Motivation and Context

Fixes https://github.com/zalando/riptide/issues/1392.    
Migration to the new mock test server is required because `com.github.restdriver.clientdriver.ClientDriver` doesn't support latest Jetty version and thus blocks migration to the Spring Boot 3 & Spring 6.
Different alternatives were analysed and discussed in the task, see https://github.com/zalando/riptide/issues/1392#issuecomment-1484743482 for more details.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation. (Not applicable as changes are in test classes only)
- [ ] I have updated the documentation accordingly. (Not applicable as changes are in test classes only)
- [ ] I have added tests to cover my changes. (Not applicable as changes are in test classes only)

### CODE REVIEW

**Comment by Semernitskaya** [comment]
I've migrated 5 modules in this draft PR:
riptide-failsafe, riptide-faults (required only removal of `rest-client-driver` dependency), riptide-httpclient, riptide-idempotency (required only removal of `rest-client-driver` dependency), riptide-logbook

File: riptide-failsafe/src/test/java/org/zalando/riptide/failsafe/MockWebServerUtil.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package org.zalando.riptide.failsafe;
```
Comment (Semernitskaya) [inline riptide-failsafe/src/test/java/org/zalando/riptide/failsafe/MockWebServerUtil.java:1]: I've collected common util methods in `MockWebServerUtil` class, currently this class is copied to the every module where I used `okhttp3.mockwebserver.MockWebServer` (the same approach is used for the existing class `MockSetup`). We can avoid copying and pasting it by introducing new project module that will be used only with test scope but I'm not sure if it is worths investing time into

Comment (fatroom) [reply riptide-failsafe/src/test/java/org/zalando/riptide/failsafe/MockWebServerUtil.java:1]: I would recommend to skip extracting separate module, totally fine with copying paste in tests

File: riptide-compression/src/test/java/org/zalando/riptide/compression/RequestCompressionPluginTest.java [hunk 2]
```diff
@@ -17,94 +19,97 @@
...
 
     @ParameterizedTest
     @ArgumentsSource(RequestFactorySource.class)
     void shouldCompressRequestBody(final ClientHttpRequestFactory factory) {
-        driver.addExpectation(onRequestTo("/")
-                        .withMethod(POST)
-                        .withHeader("X-Content-Encoding", "gzip") // written by Jetty's GzipHandler
-                        .withBody(equalTo("{}"), "application/json"),
-                giveResponse("", "text/plain"));
+        server.enqueue(textMockResponse(""));
```
Comment (Semernitskaya) [inline riptide-compression/src/test/java/org/zalando/riptide/compression/RequestCompressionPluginTest.java:56]: We cannot use Jetty's `GzipHandler` anymore, so we cannot test header inserted by it. I've tested headers inserted by the plugin and that body was actually compressed (by decompressing it and checking result)

**Comment by fatroom** [comment]
Thank you for your contribution!

**Comment by fatroom** [comment]
👍

**Pull request merged**